l2geth/core/vm/evm.go

```diff
@@ -32,10 +34,54 @@ import (
 	"github.com/ethereum/go-ethereum/rollup/dump"
 )
 
+// codec is a decoder for the return values of the execution manager. It decodes
+// (bool, bytes) from the bytes that are returned from both
+// `ExecutionManager.run()` and `ExecutionManager.simulateMessage()`
+var codec abi.ABI
+
+// innerData represents the results returned from the ExecutionManager
+// that are wrapped in `bytes`
```

Comment:
Would note somehow here that this is a convention followed by our contracts, and may not necessarily be the case for arbitrary `enqueue`s.

On the plus side, this return type will let us finally crack https://github.com/ethereum-optimism/optimism/issues/497 by adding the same return type to the `OVM_L2CrossDomainMessenger`. :)

Reply:
I think we should actually talk more about this one -- I don't think it should be necessary for us to `return abi.encode(true, bytes(""))` in `OVM_L2CrossDomainMessenger`. If we add that requirement then we're gonna get a lot of L2 contracts that weirdly return the same thing (just to make the receipts work).

Reply:
@tynes this is what I wanted to talk to you about earlier

Reply:
A bit confused, does this change imply that consuming contracts would need to follow a specific return value convention? My understanding was that this change only concerns the executionManager's returned values.

Reply:
There needs to be a strictly defined interface between the system contracts and the userland contracts for consistency purposes. Before this PR, there was not a consistent ABI - two examples are `ExecutionManger.run(...)(bytes)` and `ExecutionManager.simulateMessage(...)(bool,bytes)`. The `run()` bytes had a nested ABI encoded `(bool,bytes)`.

We want to index information about the userland contracts and not the system contracts so that when users query for receipt or transaction they get the `from` to be the ESCDA Contract Account. We enforce that a specific ABI encoding is returned from the userland contracts and passed through the system contracts such that geth can receive the data here and parse it.